api/friendsLists.js

```diff
@@ -1 +1,98 @@
-//friendsLists file - delete this if you're writing this file
\ No newline at end of file
+const express = require("express");
+const friendsRouter = express.Router();
+
+const {
+  addFriends,
+  getUserByUsername,
+  removeFriend,
+  getFriendsByUserId,
+  getPostsByUserId,
+} = require("../db");
+
+const { requireUser } = require("./utils");
+
+friendsRouter.post(
+  "/friends/:friendId",
+  requireUser,
+  async (req, res, next) => {
+    try {
+      const { username } = req.params;
+      const { id: userId } = req.user;
+      const { id: friendsId } = await getUserByUsername(username);
+
+      const addFriend = await addFriends(userId, friendsId);
+      res.send({
+        addFriend,
+        success: "You've successfully added a friend",
+      });
+    } catch ({ error, message }) {
+      next({ error, message });
+    }
+  }
+);
+
+friendsRouter.delete(
+  "/friends/:friendId",
+  requireUser,
+  async (req, res, next) => {
+    try {
+      const { username } = req.params;
+      const { id: userId } = req.user;
+      const { id: friendsId } = await getUserByUsername(username);
+
+      const deletedFriend = await removeFriend(userId, friendsId);
+      res.send({
+        deletedFriend,
+        success: "You've successfully removed a friend",
+      });
+    } catch ({ error, message }) {
+      next({ error, message });
+    }
+  }
+);
+
```

Comment:
For the above two end points we need better error handling as well. Currently someone can just send our server bad values and crash it in several different ways. Some ideas, assuming we are just going to do away with username here:

1. Error for bad user ID being provided
2. Probably want to query the friends table and make sure that these two users aren't already friends (preventive for front-end render bugs or malicious requests sent directly to our server). We probably also want to add a UNIQUE constraint on this table to prevent duplicate entries that way we can do something with a failed DB request here.